javatests/com/google/re2j/UnicodeTest.java

```diff
@@ -38,4 +39,24 @@ public void testFoldConstants() {
   // int toLower(int r);
   // int simpleFold(int r);
 
+  @Test
+  public void testOptimizedFold() {
+    // check that the new optimized fold alg will have the same result as using simple fold
+    for (int i = 0; i <= Unicode.MAX_RUNE; i++) {
+      int[] orbit = Unicode.optimizedFoldOrbit(i);
+      if (orbit != null) {
+        int r = Unicode.simpleFold(i);
```

Comment:
Hoist this statement out of the if block.

Reply:
I assumed you meant to create a method.